core/src/test/java/org/apache/calcite/test/JdbcTest.java

```diff
@@ -7401,6 +7401,20 @@ private void checkGetTimestamp(Connection con) throws SQLException {
         .runs();
   }
 
+  /**
+   * Test case for
+   * <a href="https://issues.apache.org/jira/browse/CALCITE-3894">[CALCITE-3894]
+   * The Union operation between DATE with TIMESTAMP returns a wrong result</a>.
+   */
+  @Test public void testUnionDateTime() {
+    CalciteAssert.AssertThat assertThat = CalciteAssert.that();
+    String query = "select * from (\n"
```

Comment:
between DATE with TIMESTAMP -> between DATE and TIMESTAMP ?

Reply:
Yep, thanks.